velox/functions/prestosql/FromUtf8.cpp

```diff
@@ -223,16 +223,21 @@ class FromUtf8Function : public exec::VectorFunction {
       } else {
         auto flatBase = base->asFlatVector<StringView>();
         auto stringBuffers = flatBase->stringBuffers();
-        localResult = decodedInput.wrap(
-            std::make_shared<FlatVector<StringView>>(
-                context.pool(),
-                VARCHAR(),
-                nullptr,
-                flatBase->size(),
-                flatBase->values(),
-                std::move(stringBuffers)),
-            *input,
-            rows.end());
+
+        auto values = AlignedBuffer::allocate<StringView>(
+            rows.end(), context.pool(), StringView());
+        auto* rawValues = values->asMutable<StringView>();
+        rows.applyToSelected([&](auto row) {
+          rawValues[row] = decodedInput.valueAt<StringView>(row);
+        });
+
+        localResult = std::make_shared<FlatVector<StringView>>(
```

Comment:
might be worth trying out PeeledEncoding here since usage pattern here is similar to peeling and eventually wrapping. toVarcharNoCopy() can take in the actual input (not the decoded one). PeeledEncoding would then return a flat or constant vector if it was successful, then we can use PeeledEncoding::wrap() to wrap it at the end. wrap() ensures that only the active rows are marked as non-null so this bug would be fixed

Reply:
Not sure if the complexity is worth it.

Reply:
do you mean complexity in terms of extra code in the peeled encoding implementation being used or in terms of the coded added in toVarcharNoCopy() to use PeeledEncoding.
I am trying to get an idea if there is anything I can improve in the API to make it a better choice for use-cases like we have here. The intent here is to avoid using DecodedVectors's 3 most tricky functions since they are the most prone to bugs like what happened here.(https://github.com/facebookincubator/velox/blob/main/velox/vector/DecodedVector.h#L231-L289 , I wanted to remove these but could not due how DecodedVector is structured and how peeledEncodng relies on it)